src/adr/0041-james-servers-offering.md

```diff
@@ -0,0 +1,159 @@
+for storing data, but cannot be used in a clustered setup, as there is no messaging happening between nodes, which 
+prevents some James features to work correctly. This server furthermore is extensible, and ships many extra features 
+by default.
+
+ - **Distributed James** (`server/container/guice/cassandra-rabbitmq-guice`) and its LDAP counter part
+(`server/container/guice/cassandra-rabbitmq-ldap-guice`) enabled to achieve the distributed use case. This
+server furthermore is extensible, and ships many extra features by default. It requires good knowledge of 
+underlying services like Cassandra, S3-like storage, RabbitMQ and ElasticSearch.
+
+ - **Guice memory** (`server/container/guice/memory-guice`) ships a server based on memory storage. Easy to boot, 
+without external dependencies, we mostly rely on it for testing purposes.
+
+ - **JPA Guice** (`server/container/guice/jpa-guice`) ships an extensible server based on JPA to store data in 
+a RDBMS. This server has no external dependencies by default. However the still large amount of configuration 
+can make new comers confused. It currently satisfies operator looking for self hosting solutions as well as 
+those looking for an extensible server.
+
+ - **JPA SMTP** (`server/container/guice/jpa-smtp`) demonstrates a Mail Transfer Agent, dedicated to mail
+processing, that does not ship the James Mailbox, as well as Mail Delivery protocols like IMAP, JMAP or POP3.
+
+We can notice that this offer is specified in technical terms, and does not directly map to identified user needs.
+
+Also the large amount of servers makes the choice harder to make for newcomers.
+
+## Decision
+
+We should deliver a subset of mail servers directly addressing user needs and brand them according to their use cases,
+not their technical details.
+
+In order to arrive at a clearer offering we should first reduce the server cardinality. This can be achieved via:
+
+ - **Spring server** deprecation and removal
+ - Merging Guice servers with their LDAP variations (see [ADR 36 about guice module choosing](0036-against-use-of-conditional-statements-in-guice-modules.md))
+
+We should then rename our servers in order to match the use cases:
+
+ - **Basic server** addresses the self-hosting use case. It will be based on current *jpa-guice* server but with the clear intent to 
+lower adoption barriers. It exposes limited configurations in order not to confuse the user. Configuration and documentation is centered 
+around self hosting use cases. Configuration and documentation addresses the self-hosting usage without exposing underlying James concepts.
+
+The **Basic Server** is a subset of the advanced server, and differs from it from documentation and configuration. Users of the basic server feeling
+constraint by its minimalistic configuration capabilities and features would then be able to easily upgrade to the advanced server.
```

Comment:
When working on the Basic Server, I realized that it may be useful to allow a concept of "modes". I.e., either it is a fixed product (you get SMTP, IMAP, etc. with no choice), or you can set it to "Relay Only Mode", "Full Mode" etc.

Not sure if this is useful information here, or if it changes anything relating to this document, but thought I'd mention it just in case.

Reply:
The "Basic server" is intended for self hosting so should target "SMTP and IMAP modes" while not addressing the others for simplicity?

Reply:
Yes, agree that simplicity is the primary objective here.